internal/pipeline/runtime/container.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package runtime
+
+// some of these codes are copied from
+// - https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/fn/runtime/container/container.go
+// - https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/fn/runtime/exec/exec.go
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/user"
+	"time"
+)
+
+// containerNetworkName is a type for network name used in container
+type containerNetworkName string
+
+const (
+	networkNameNone containerNetworkName = "none"
+	networkNameHost containerNetworkName = "host"
+	defaultTimeout  time.Duration        = 5 * time.Minute
+)
+
+// ContainerFnPermission contains the permission of container
+// function such as network access.
+type ContainerFnPermission struct {
+	AllowNetwork  bool
+	AsCurrentUser bool
+}
+
+// ContainerFn implements a KRMFn which run a containerized
+// KRM function
+type ContainerFn struct {
+	// Image is the container image to run
+	Image string
+	// Container function will be killed after this timeour.
+	// The default value is 5 minutes.
+	Timeout time.Duration
+	Perm    ContainerFnPermission
+}
+
+// Run implements KRMFn. It will run the container function with
+// stdin from r and write the output to w
+func (f *ContainerFn) Run(r io.Reader, w io.Writer) error {
```

Comment:
Are you implementing this to work on Windows?

Reply:
@droot Do we want to support Windows?

Reply:
That's a good question. Didn't think about this. We would like to support windows, but reality is testing it regularly requires access to windows machines etc and I have found that to be challenging. Actual work required to support it won't be much IMO. From my previous project experience, what has helped is to create an issue for windows support and our friends from Windows eco-system help to support that.

TL;DR: windows support will be best effort basis and not a priority at this point.

Reply:
I think we should avoid regression if kpt supports windows today, but it can be punted for now and tracked as a support issue.

Reply:
https://github.com/GoogleContainerTools/kpt/issues/1367